packages/engine/Source/Scene/MetadataPicking.js

```diff
@@ -262,11 +263,52 @@ MetadataPicking.convertToObjectType = function (type, value) {
   return value;
 };
 
+/**
+ * Converts the given type into an raw value or array representation.
+ *
+ * For `VECn/MATn` types, the given value is converted into an array.
+ * For other types, the value is returned directly
+ *
+ * @param {string} type The `ClassProperty` type
+ * @param {any} value The input value
+ * @returns {any} The array representation
+ */
+MetadataPicking.convertFromObjectType = function (type, value) {
+  if (!defined(value)) {
+    return value;
+  }
+  if (
+    type === MetadataType.SCALAR ||
+    type === MetadataType.STRING ||
+    type === MetadataType.BOOLEAN ||
+    type === MetadataType.ENUM
+  ) {
+    return value;
+  }
+  switch (type) {
+    case MetadataType.VEC2:
+      return Cartesian2.pack(value, Array(2));
+    case MetadataType.VEC3:
+      return Cartesian3.pack(value, Array(3));
+    case MetadataType.VEC4:
+      return Cartesian4.pack(value, Array(4));
+    case MetadataType.MAT2:
+      return Matrix2.unpack(value, Array(4));
+    case MetadataType.MAT3:
+      return Matrix3.unpack(value, Array(9));
+    case MetadataType.MAT4:
+      return Matrix4.unpack(value, Array(16));
```

Comment:
The `unpack` was a copy-and-paste error, and it did not show up due to the limitations of the types that can appear in property textures in general.

More broadly, this is related to https://github.com/CesiumGS/cesium/pull/12075#discussion_r1781538033 : It's unfortunate that it was necessary to create these back-and-forth conversion functions _here_ . This PR (and the linked one) _should_ have _used_ existing functions (which _should_ already be covered by specs).

But ... ignoring the spilled milk: If course, the current coverage of the `MetadataPicking` is dismal, and I started creating specs for that. Given the combinatorial explosion, this is ongoing, and will at some point have to be aligned with the existing structures, as mentioned in the comment that I linked to above.

Reply:
Thanks. Some tests are better than no tests!

Reply:
I added (more) specs, and tried to handle the combinatorial explosion by being exhaustive in the "low-level" functions, and picking "representative cases" in the higher-level ones. 

The result is currently https://github.com/CesiumGS/cesium/blob/49bd80e6aea84b6f467e5fca9df6d9c52ebd66e5/packages/engine/Specs/Scene/MetadataPickingSpec.js 

I've seen few cases with nested `describe`s, as in
```
describe("The class...") {
    describe("the function...") {
       it("does this" ...),
       it("does that" ...),
    }
}
```
but thought that this could make sense here (and ... in many other classes, actually...)

If this should be "flattened", just drop me a note.

Reply:
Nested defines are all good.